cmake/detect-install-dirs.cmake

```diff
@@ -24,6 +27,20 @@ if (DEFINED INC_INSTALL_DIR)
     set(CMAKE_INSTALL_INCLUDEDIR "${INC_INSTALL_DIR}")
 elseif (DEFINED INSTALL_INC_DIR)
     set(CMAKE_INSTALL_INCLUDEDIR "${INSTALL_INC_DIR}")
+elseif (CMAKE_INSTALL_PREFIX STREQUAL "/boot/system")
+    # Haiku system headers
+    set(CMAKE_INSTALL_INCLUDEDIR "develop/headers")
+endif()
+
+# Determine installation directory for man files
+if (DEFINED MAN_INSTALL_DIR)
+    set(MAN_INSTALL_DIR "${MAN_INSTALL_DIR}" CACHE PATH "Installation directory for manuals (Deprecated)" FORCE)
+    set(CMAKE_INSTALL_MANDIR "${MAN_INSTALL_DIR}")
+elseif (DEFINED INSTALL_MAN_DIR)
+    set(CMAKE_INSTALL_MANDIR "${INSTALL_MAN_DIR}")
+elseif (CMAKE_INSTALL_PREFIX STREQUAL "/boot/system")
+    # Haiku system manuals
+    set(CMAKE_INSTALL_MANDIR "documentation/man")
```

Comment:
Why do we need to add ``MAN_INSTALL_DIR`` and ``INSTALL_MAN_DIR`` if they were not used before, instead of using the standard ``CMAKE_INSTALL_MANDIR`` variable?


``MAN_INSTALL_DIR`` has been removed as unused: https://github.com/zlib-ng/zlib-ng/pull/885

Reply:
This is just to document all the directories, so I can run pkgcheck and see it pass... There is still a lot of unused code in both configure and CMakeLists.txt for backwards compatibility.